dbms/src/Flash/Planner/plans/PhysicalProjection.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Logger.h>
+#include <DataStreams/ExpressionBlockInputStream.h>
+#include <Flash/Coprocessor/DAGContext.h>
+#include <Flash/Coprocessor/DAGExpressionAnalyzer.h>
+#include <Flash/Coprocessor/DAGPipeline.h>
+#include <Flash/Coprocessor/InterpreterUtils.h>
+#include <Flash/Planner/FinalizeHelper.h>
+#include <Flash/Planner/PhysicalPlanHelper.h>
+#include <Flash/Planner/plans/PhysicalProjection.h>
+#include <Interpreters/Context.h>
+
+namespace DB
+{
+PhysicalPlanPtr PhysicalProjection::build(
+    const Context & context,
+    const String & executor_id,
+    const LoggerPtr & log,
+    const tipb::Projection & projection,
+    const PhysicalPlanPtr & child)
+{
+    assert(child);
```

Comment:
I'm not sure if we recommend to use RUNTIME_ASSERT instead of assert for such cases.

Reply:
Since this will not be caused by external input, I think either is fine.